a part of some classes added

Niktoya45 opened the pull request.

Niktoya45 committed:
fixes

Niktoya45 committed:
a part of db manipulation classes added

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Api/Application/Commands/CreateUserTaskCommand.cs:
```diff
@@ -5,27 +5,17 @@ namespace TrialsSystem.UserTaskService.Api.Application.Commands
             Dictionary<string, string> additionalProperties)
         {
-            Status = status;
+            UserId = userId;
```
1 m-w us-p

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Api/Controllers/v1/UserTaskController.cs:
```diff
@@ -34,7 +34,7 @@ public UserTaskController(IMediator mediator)
-        public async Task<IActionResult> GetAsync(
+        public async Task<IActionResult> GetListAsync(
             [FromRoute] string userId,
             [FromQuery] int? skip = 0,
```
2 q c-s

cherryarft replied:
IUserProfileProvider { UserProfile Get(); void Set(); }
middleware{ .. IUserProfileProvider, Invoke() { <userId> -> upp.Set(userId) } }
...

*Handler() { ctor(IUserProfileProvider) .. Handle(){  var userId = upp.get() } } 
.... 

service.addScope<IUserProfileProvider>

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Api/Middlewares/ExceptionUserTaskMiddleware.cs:
```diff
@@ -26,6 +27,10 @@ public async Task Invoke(HttpContext context, RequestDelegate next)
                         context.Response.StatusCode = StatusCodes.Status400BadRequest;
                         break;
 
+                    case DomainException de:
+                        context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                        break;
```
3 ? e-s

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Domain/AggregatesModel/Base/Entity.cs:
```diff
@@ -0,0 +1,13 @@
+        public bool IsDeleted { get; protected set; }
+
+        public DateTime CreatedDate { get; protected set; }
+        public DateTime LastModifiedDate { get; protected set; }
```
4 n-l ?

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Domain/AggregatesModel/UserTaskAggregate/UserTask.cs:
```diff
@@ -0,0 +1,65 @@
+
+            Name = name;
+
+            CreatedDate = DateTime.UtcNow;
```
5 i-r

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Domain/AggregatesModel/UserTaskAggregate/UserTask.cs:
```diff
@@ -0,0 +1,65 @@
+            AdditionalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        public void SetStatus(string statusName)
```
6 h_s

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Domain/AggregatesModel/UserTaskAggregate/UserTaskStatus.cs:
```diff
@@ -0,0 +1,11 @@
+﻿
+namespace TrialsSystem.UserTaskService.Domain.AggregatesModel.UserTaskAggregate
+{
+    public enum UserTaskStatus
```
7 inf

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Domain/Exceptions/InvalidTaskStatusChangeException.cs:
```diff
@@ -0,0 +1,14 @@
+    /// Used when current task status cannot be changed 
+    /// from current to the given valid status
+    /// </summary>
+    public class InvalidTaskStatusChangeException : DomainException
+    {
+        public InvalidTaskStatusChangeException(string st1, string st2) : base($"Cannot change task status from {st1} to {st2}")
+        {
+        }
+    }
```
8 u-_-

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Context/UserTaskDbContext.cs:
```diff
@@ -0,0 +1,28 @@
+    {   
+        public DbSet<UserTask> UserTasks { get; init; }
+
+        private readonly string _connectionStr = @"mongodb://localhost:27017";
```
9 cfg

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Models/UserTaskDTOs/UpdateUserTaskRequest.cs:
```diff
@@ -4,11 +4,9 @@ public class UpdateUserTaskRequest
+        public string UserId { get; set; }
 
-        public DateTime LastUpdatedDateTime { get; set; }
+        public string Status { get; set; }
```
10 cfg str-em

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Repositories/Abstractions/IRepository.cs:
```diff
@@ -0,0 +1,17 @@
+
+namespace TrialsSystem.UserTaskService.Infrastructure.Repositories.Abstractions
+{
+    public interface IRepository<T>
```
11 w-r r-r

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Repositories/Implementations/UserTaskRepository.cs:
```diff
@@ -0,0 +1,63 @@
+    {
+        private UserTaskDbContext _dbcontext;
+
+        FilterDefinition<UserTask> filterDeleted = Builders<UserTask>.Filter.Where(ut => !ut.IsDeleted);
+
```
.. i-r

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Repositories/Implementations/UserTaskRepository.cs:
```diff
@@ -0,0 +1,63 @@
+
+        public async Task<UserTask> Add(UserTask task, CancellationToken ct = default)
+        {
+            await _dbcontext.UserTasks.AddAsync(task, ct);
```
.. f-t

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Infrastructure/Repositories/Implementations/analyzerlog.sarif:
```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-1.0.0",
```
.. ? mm

cherryarft commented on src/UsersManagement/TrialsSystem.UsersService/TrialsSystem.UsersService.Domain/AggregatesModel/Base/ValueObject.cs:
```diff
@@ -0,0 +1,56 @@
+
+namespace TrialsSystem.UsersService.Domain.AggregatesModel.Base
+{
+    public abstract class ValueObject
```
.. m-k

cherryarft commented on src/UsersManagement/TrialsSystem.UsersService/TrialsSystem.UsersService.Domain/AggregatesModel/DeviceAggregate/Device.cs:
```diff
@@ -1,21 +1,31 @@
+	
+	    private string DeviceTypeId { get; set; }
+	
+	    [ForeignKey("DeviceTypeId")]
+        public virtual DeviceType Type { get; private set; }
```
.. f-c

cherryarft commented on src/UsersManagement/TrialsSystem.UsersService/TrialsSystem.UsersService.Domain/AggregatesModel/DeviceAggregate/DeviceType.cs:
```diff
@@ -2,9 +2,14 @@
         public string Id { get; set; }
         public string Name { get; set; }
+
+        protected override IEnumerable<object> GetEqualityComponents() {
```
.. r-r-d

cherryarft commented on src/UsersManagement/TrialsSystem.UsersService/TrialsSystem.UsersService.Domain/AggregatesModel/UserAggregate/User.cs:
```diff
@@ -1,21 +1,18 @@
     public class User : Entity
     {
-        public User(string email, string name, string surname, string cityd, string genderId, DateTime birthDate)
+        public User(string email, string name, string surname, string cityId, string genderId, DateTime birthDate)
```
.. ? c-r _ f

cherryarft commented on src/UsersManagement/TrialsSystem.UsersService/TrialsSystem.UsersService.Infrastructure/Repositories/Abstractions/IRepository.cs:
```diff
@@ -0,0 +1,16 @@
+﻿
+namespace TrialsSystem.UsersService.Infrastructure.Repositories.Abstractions
+{
+    public interface IRepository<T>
```
.. ? 2s

cherryarft commented on src/UsersManagement/TrialsSystem.UserTaskService/TrialsSystem.UserTaskService.Api/Controllers/v1/UserTaskController.cs:
```diff
@@ -34,7 +34,7 @@ public UserTaskController(IMediator mediator)
-        public async Task<IActionResult> GetAsync(
+        public async Task<IActionResult> GetListAsync(
             [FromRoute] string userId,
             [FromQuery] int? skip = 0,
```
skip&take -> PageQuery { Skip, Take }

Niktoya45 committed:
UserTaskService mostly implemented

things yet to do:
      *  -- managing business logic exceptions with State prty change;
      *  -- reviewing controller's methods implementation;
      *  -- finishing UserService with UoW, Interceptors, Profiles, Mappings, Handlers

Niktoya45 committed:
Command/Query Handlers implemented

implemented UoW, Repositories, Handlers;
minor changes in UserTask Controller and handler methods;
